datahub/dnb_api/test/test_tasks.py

```diff
@@ -821,3 +824,340 @@ def test_update_company_from_dnb_data_fails_validation(dnb_response_uk, caplog):
+            'fields_to_update': ['global_ultimate_duns_number'],
+            'dnb_modified_on_before': (now() + timedelta(days=1)).strftime('%Y-%m-%dT%H:%M:%S'),
+            'simulate': False,
+            'limit': 1,
+        },
+    )
+
+    assert task_result.successful()
+    company_1.refresh_from_db()
+    company_2.refresh_from_db()
+    # We expect company_1 to be unmodified
+    assert company_1.dnb_modified_on == original_company_1.dnb_modified_on
+    # We expect company_2 to be modified, as it was least recently synced with D&B
+    assert company_2.dnb_modified_on == now()
+
+
+@freeze_time('2019-01-01 11:12:13')
+def test_sync_outdated_companies_limit_most_recently_interacted_updated(
+    requests_mock,
+    dnb_response_uk,
+):
+    """
+    Test that running sync_outdated_companies_with_dnb with a limit will update
+    the most recently interacted company.
+    """
+    requests_mock.post(
+        DNB_SEARCH_URL,
+        json=dnb_response_uk,
+    )
+
+    company_most_recent_interaction = CompanyFactory(
+        duns_number='123456789',
+        dnb_modified_on=now() - timedelta(days=1),
+    )
+    CompanyInteractionFactory(company=company_most_recent_interaction, date=now())
+
+    company_least_recent_interaction = CompanyFactory(
+        duns_number='123456788',
+        dnb_modified_on=now() - timedelta(days=1),
+    )
+    CompanyInteractionFactory(
+        company=company_least_recent_interaction,
+        date=now() - timedelta(days=1),
+    )
+
+    task_result = sync_outdated_companies_with_dnb.apply_async(
+        kwargs={
+            'fields_to_update': ['global_ultimate_duns_number'],
+            'dnb_modified_on_before': (now() + timedelta(days=1)).strftime('%Y-%m-%dT%H:%M:%S'),
+            'simulate': False,
+            'limit': 1,
+        },
+    )
+
+    company_least_recent_interaction.refresh_from_db()
+    company_most_recent_interaction.refresh_from_db()
+    assert task_result.successful()
+    # We expect the least recently interacted company to be unmodified
+    assert company_least_recent_interaction.dnb_modified_on == now() - timedelta(days=1)
+    # We expect most recently interacted company to be modified
+    assert company_most_recent_interaction.dnb_modified_on == now()
+
+
+@freeze_time('2019-01-01 11:12:13')
+def test_sync_outdated_companies_nothing_to_update(
+    requests_mock,
+    dnb_response_uk,
+):
+    """
+    Add two companies (one with dnb_modified_on>dnb_modified_on_before) and
+    assert that only the outdated one is synced.
+    """
+    company = CompanyFactory(
+        duns_number='123456789',
+        dnb_modified_on=now() - timedelta(days=5),
+    )
+    original_company = Company.objects.get(id=company.id)
+
+    task_result = sync_outdated_companies_with_dnb.apply_async(
+        kwargs={
+            'fields_to_update': ['global_ultimate_duns_number'],
+            'dnb_modified_on_before': (now() - timedelta(days=1)).strftime('%Y-%m-%dT%H:%M:%S'),
```

Comment:
Are we able to use ISO format instead of using a custom format string?
```suggestion
            'dnb_modified_on_before': (now() - timedelta(days=1)).isoformat(),
```

Reply:
I also think it would be great if we could use `isoformat`.

Reply:
Another option would be a relative value e.g. number of days ago.

Reply:
I just thought about it and realised that I was hung up with making the `dnb_modified_on_before` argument a string for no reason.  I've changed it to expect a datetime instead which allows us to do away with this whole conversion business - thanks for making me think about this properly!

Reply:
Looks like Celery serialises it in ISO format automatically – works out fine though, so seems good to me.